### DISCUSSION TITLE
Chapter 12 - Bug Using Softmax Output on CrossEntropyLoss

### DISCUSSION THREAD

**Discussion by liereynaldo** [rasbt/machine-learning-book#222] [General]
![image](https://github.com/user-attachments/assets/5f0d0f78-1eaa-4b85-a446-52abaa6156ab)

Since CrossEntropyLoss on Pytorch expecting Logit input is incompatible with Softmax outputting probability, the loss will be Loss=−log(Softmax(Softmax(logits))), so to resolve this either: 
1) The last layer of the model (softmax) should be removed & keep CrossEntropyLoss 
2) Change the last layer into logsoftmax and use NLLLoss as loss function
